plasticos_intake_normalizer/models/intake_normalizer.py

```
@@ -0,0 +1,501 @@
+            errors.append({
+                "field": "partner_id",
+                "message": "Partner is required.",
+                "code": "INT-004",
+            })
+
+        # ── Range validations (non-blocking if field empty) ──
+        if config.validate_density_range and self.density_value:
+            if not (0.5 <= self.density_value <= 2.5):
+                errors.append({
+                    "field": "density_value",
+                    "message": f"Density {self.density_value} outside valid range 0.5–2.5 g/cm³.",
```

Ruff (0.15.1)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)